sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/Offset.scala

```diff
@@ -17,11 +17,14 @@
 
 package org.apache.spark.sql.execution.streaming
 
+import com.fasterxml.jackson.annotation.JsonTypeInfo
+
 /**
  * An offset is a monotonically increasing metric used to track progress in the computation of a
  * stream. An [[Offset]] must be comparable, and the result of `compareTo` must be consistent
  * with `equals` and `hashcode`.
  */
+@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.PROPERTY, property = "@class")
```

Comment:
What does this mean?  What JSON does this produce?

@vanzin, I find this very hard to reason about relative to constructing the JSON by hand.